crates/typed-store/src/rocks/iter.rs

```diff
@@ -68,4 +75,35 @@ impl<'a, K: Serialize, V> Iter<'a, K, V> {
         self.db_iter.seek_to_last();
         self
     }
+
+    /// Will make the direction of the iteration reverse and will
+    /// create a new `RevIter` to consume. Every call to `next` method
+    /// will give the next element from the end.
+    pub fn reverse(mut self) -> RevIter<'a, K, V> {
+        self.direction = Direction::Reverse;
+        RevIter::new(self)
+    }
+}
+
+/// An iterator with a reverted direction to the original. The `RevIter`
+/// is hosting an iteration which is consuming in the opposing direction.
+/// It's not possible to do further manipulation (ex re-reverse) to the
+/// iterator.
+pub struct RevIter<'a, K, V> {
```

Comment:
Introduced a separate struct so can limit the user from doing any complex combinations of seeks that could confuse the iterator. Now there is only one option once you reverse...you are calling `next` to traverse the results backwards